src/werkzeug/exceptions.py

```diff
@@ -600,11 +600,29 @@ class TooManyRequests(HTTPException):
     to identify users and their request rates). The server may include a
     "Retry-After" header to indicate how long the user should wait before
     retrying.
+
+    .. versionchanged:: 0.16.1
+        ``retry_after_secs`` was added as the first argument, ahead of
+        ``description``.
     """
 
     code = 429
     description = "This user has exceeded an allotted request count. Try again later."
 
+    def __init__(self, description=None, retry_after_secs=None):
+        """
+        Use the optional value of retry_after_secs to specify the number of seconds
+        to wait for a retry attempt.
+        """
+        HTTPException.__init__(self, description)
+        self.retry_after_secs = retry_after_secs
```

Comment:
Looks like the `Retry-After` could also be a [HTTP-date](https://tools.ietf.org/html/rfc7231#section-7.1.3), so maybe it should accept `retry_after: Union[datetime, int]`, so something like?
```python
if isinstance(retry_after, datetime):
    self.retry_after = http_date(retry_after)
else:
    self.retry_after = "{0:.0f}".format(retry_after)  # Must be digits only
```